crates/provider-blobstore-nats/src/blobstore.rs

```diff
@@ -0,0 +1,678 @@
+                .create_object_store(container_config)
+                .await
+                .context("failed to create NATS Blobstore Container")
+                .map(|_| ())
+        }
+        .await
+        .map_err(|err| format!("{err:#}")))
+    }
+
+    // Get metadata of an existing NATS Blobstore Container
+    #[instrument(level = "trace", skip(self))]
+    async fn get_container_info(
+        &self,
+        context: Option<Context>,
+        name: String,
+    ) -> anyhow::Result<Result<bindings::wrpc::blobstore::types::ContainerMetadata, String>> {
+        Ok(async {
+            propagate_trace_for_ctx!(context);
+
+            // Retrieve the blobstore connection
+            let blobstore = self
+                .get_blobstore(context)
+                .await
+                .context("failed to get NATS Blobstore connection")?;
+
+            // Attempt to get the container metadata
+            let _container = blobstore
+                .jetstream
+                .get_object_store(name)
+                .await
+                .context("failed to get container info")?;
+
+            // Construct and return the container metadata
+            let metadata = bindings::wrpc::blobstore::types::ContainerMetadata {
+                created_at: 0u64, // Unix epoch as a placeholder
+            };
+            Ok(metadata)
+        }
+        .await
+        .map_err(|err: anyhow::Error| err.to_string()))
+    }
+
+    // Check if a NATS Blobstore Container exists
+    #[instrument(level = "debug", skip(self))]
+    async fn container_exists(
+        &self,
+        context: Option<Context>,
+        name: String,
+    ) -> anyhow::Result<Result<bool, String>> {
+        Ok(async {
+            propagate_trace_for_ctx!(context);
+
+            // Retrieve the NATS blobstore connection
+            let blobstore = self
+                .get_blobstore(context)
+                .await
+                .map_err(|e| e.to_string())?;
+
+            // Check if the container exists
+            match blobstore.jetstream.get_object_store(&name).await {
+                Ok(_) => Ok(true),
+                Err(e)
+                    if matches!(
+                        e.kind(),
+                        async_nats::jetstream::context::ObjectStoreErrorKind::GetStore
+                    ) =>
+                {
+                    Ok(false)
+                }
+                Err(e) => Err(format!("failed to check container existence: {e}")),
+            }
+        }
+        .await
+        .map_err(|err| err.to_string()))
+    }
+
+    /// Retrieve data from an object in the specified NATS blobstore Container
+    /// Optionally specify start and end byte offsets for partial reads
+    #[instrument(level = "debug", skip(self))]
+    async fn get_container_data(
+        &self,
+        context: Option<Context>,
+        id: bindings::wrpc::blobstore::types::ObjectId,
+        start: u64,
+        end: u64,
```

Comment:
it doesn't seem like the `start` and `end` parameters are being used. See this one as an example: https://github.com/wasmCloud/wasmCloud/blob/main/crates/provider-blobstore-s3/src/lib.rs#L790

Reply:
NATS doesn't support reading ranges; that's why in all NATS providers the range is not being used.